### PR TITLE
Updates toolchain to a more recent 2023-03-25

### PR DESCRIPTION
**Pull request by mpapierski**
This bumps a toolchain for contract API to more recent 2023-03-25. This is not the most recent, as the current nightly appears to produce non-mvp WebAssembly opcodes which are not supported on our platform. Apparently this version update also decerases gas cost on average in our tests.

See also upstream compiler issue here: https://github.com/rust-lang/rust/issues/109807

### CODE REVIEW

**Review by gRoussac — APPROVED** [review]

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

File: execution_engine_testing/tests/src/test/explorer/faucet.rs [hunk 1]
```diff
@@ -841,13 +841,13 @@ fn faucet_costs() {
...
     // without understanding why the cost has changed.  If the costs do change, it should be
     // reflected in the "Costs by Entry Point" section of the faucet crate's README.md.
-    const EXPECTED_FAUCET_INSTALL_COST: u64 = 71_691_403_720;
-    const EXPECTED_FAUCET_SET_VARIABLES_COST: u64 = 623_279_550;
-    const EXPECTED_FAUCET_CALL_BY_INSTALLER_COST: u64 = 3_101_326_690;
-    const EXPECTED_FAUCET_CALL_BY_USER_COST: u64 = 3_350_546_610;
+    const EXPECTED_FAUCET_INSTALL_COST: u64 = 75_226_865_930;
+    const EXPECTED_FAUCET_SET_VARIABLES_COST: u64 = 579_464_060;
+    const EXPECTED_FAUCET_CALL_BY_INSTALLER_COST: u64 = 3_040_141_320;
+    const EXPECTED_FAUCET_CALL_BY_USER_COST: u64 = 3_242_199_690;
```
Comment (Fraser999) [inline execution_engine_testing/tests/src/test/explorer/faucet.rs:847]: As per the comment above these consts, could you update the table in the indicated README too please?

**Review by Fraser999 — CHANGES_REQUESTED** [review]

File: execution_engine_testing/tests/src/test/explorer/faucet.rs (hunk 1 above)
Comment (mpapierski) [reply execution_engine_testing/tests/src/test/explorer/faucet.rs:847]: Done

**Review by Fraser999 — APPROVED** [review]

**Pull request merged**